### ISSUE TITLE
Order matters when stitching call graphs with LocalMerger

### ISSUE THREAD

**Issue by cg122** [fasten-project/fasten#252]
## Order matters when stitching call graphs with LocalMerger.mergeAllDeps

The order that adds dependency call graphs to local merger has impact of stitching result.

## To Reproduce
See test case [mergeAllDepsOrderMattersTest()](https://github.com/fasten-project/fasten/blob/7737cdb551b700475f78ae1958503eb65121bf65/core/src/test/java/eu/fasten/core/merge/LocalMergerTest.java#L65) in LocalMergerTest 

```
        var merger12 = new LocalMerger(Arrays.asList(first, second));
        var cg12 = merger12.mergeAllDeps();
        var merger21 = new LocalMerger(Arrays.asList(second, first));
        var cg21 = merger21.mergeAllDeps();
        assertEquals(cg12.numArcs(),cg21.numArcs());
        assertEquals(cg12.numNodes(),cg21.numNodes());
```

The different order adding dependencies results in the difference of total nodes in the stitched graph.
```
org.opentest4j.AssertionFailedError: 
Expected :959
Actual   :945
```
## Expected behavior
Feature or bug?

## Additional context
The dependency call graphs (hamcrest-2.2.json and hamcrest-core-1.3.json) in the test contain callables with identical local FASTEN uris. These libraries also trigger stitching [issue](https://github.com/fasten-project/fasten-maven-plugin/issues/2) in fasten-maven-plugin.

**Comment by cg122** [comment]
It seems also the case for fasten-maven-plugin @tmortagne .

**Comment by cg122** [comment]
For the new version of call graph generated, order issue still holds. It can be reproduced with newly generated call graphs in the same test case.

**Comment by ashkboos** [comment]
@cg122 this should be fixed in PR #255. Also, note that ERCGs are mutable objects so you can only merge them once. For example, in your test, you need to load them twice from a file or somehow do a deep copy of them if you need to merge twice.
